azure-quantum/azure/quantum/workspace.py

```diff
@@ -132,6 +135,7 @@ def __init__(
         location: Optional[str] = None,
         credential: Optional[object] = None,
```

Comment:
This is typed way too loosely. Are you really accepting *any* kind of object here?